Added functions to make compatible with polarization calculations

AnnieStephenson opened the pull request:
added functions and tests for Mie calculations in Cartesian lab reference frame so we can calculate scattered polarizations in the lab frame

AnnieStephenson commented:
_xy functions have now been combined with their corresponding functions in the 'scattering plane' coordinate system. _xy is now generally referred to as 'cartesian'. The user can now specify a coordinate system using an optional parameter. Docstrings have been updated to reflect the changes. Tests have also been updated.

vnmanoharan commented:
My preference is to have one function that handles both scalar and vector inputs rather than two separate functions.  The function should inspect the arguments to see if they are scalars or vectors and act accordingly.  In other words, the function should act like a Numpy ufunc.  This approach eliminates duplication of code, so that if we need to change an algorithm, we can do it in only one place rather than two.

AnnieStephenson commented:
The vectorized functions handle both scalar and vector inputs, and I've now removed the functions that took a scalar only. I've also replaced the calls to the scalar functions throughout the code and updated the tests.

vnmanoharan commented on pymie/mie.py:
```diff
@@ -646,41 +738,87 @@ def integrate_intensity_complex_medium(I_par, I_perp, distance, theta, k,
+    dsigma_1: ndarray (in units of length**2)
+        differential cross section for first component of basis
+    dsigma_2: ndarray (in units of length**2)
+        differential cross section for second component of basis
```
I'm a little confused about why this function returns a differential scattering cross section.  The docs (and function name) say it returns the integral.

AnnieStephenson replied:
This function has always returned the differential scattering cross section in addition to the integrated cross section. This was originally written this way for testing purposes and should probably now be removed. However, since montecarlo.py relies on this function and expects a tuple of length 5, I think it would be better to change this as a part of the refactoring. I can raise an issue so we don't forget to fix this.

vnmanoharan commented on pymie/mie.py:
```diff
@@ -785,22 +923,199 @@ def diff_abs_intensity_complex_medium(m, x, theta, ktd):
+     
     
-def _amplitude_scattering_matrix(n_stop, prefactor, coeffs, theta):
+    see pages 35,51-53 in Annie Stephenson lab notebook #3 for details
```
Let's file an issue to put the relevant info in the documentation, once we have documentation.  It will be much easier to look up there than to find your lab notebook.

vnmanoharan commented:
minor changes requested in code review (see above)
